### PR TITLE
add new package: glusterfs

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/glusterfs/package.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+class Glusterfs(AutotoolsPackage):
+    """Gluster is a software defined distributed storage that can scale to
+    several petabytes. It provides interfaces for object, block and file
+    storage."""
+
+    homepage = "https://www.gluster.org/"
+    url      = "https://download.gluster.org/pub/gluster/glusterfs/7/7.3/glusterfs-7.3.tar.gz"
+    list_depth = 1
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/glusterfs/package.py]: I think this should be:
```python
list_url = "https://download.gluster.org/pub/gluster/glusterfs/"
list_depth = 2
```
You can test this with `spack versions glusterfs`.

Comment (darmac) [reply var/spack/repos/builtin/packages/glusterfs/package.py]: Fix it in: https://github.com/spack/spack/pull/15565/commits/020ddc51e659b7a3d46b92e40b6a49fc863cdc11

File: var/spack/repos/builtin/packages/glusterfs/package.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+class Glusterfs(AutotoolsPackage):
+    """Gluster is a software defined distributed storage that can scale to
+    several petabytes. It provides interfaces for object, block and file
+    storage."""
+
+    homepage = "https://www.gluster.org/"
+    url      = "https://download.gluster.org/pub/gluster/glusterfs/7/7.3/glusterfs-7.3.tar.gz"
+    list_depth = 2
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/glusterfs/package.py]: You still need a `list_url`: https://spack.readthedocs.io/en/latest/packaging_guide.html#attribute-list-url

Comment (darmac) [reply var/spack/repos/builtin/packages/glusterfs/package.py]: I have overwrited url_for_version. 
maybe I should just remove `list_depth` and no need for `list_url` definition?
```
    def url_for_version(self, version):
        url = 'https://download.gluster.org/pub/gluster/glusterfs/{0}/{1}/glusterfs-{1}.tar.gz'
        return url.format(version.up_to(1), version)
```
I can stage all versions between 6.7 to 7.3 by using `url_for_version()`

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/glusterfs/package.py]: `url_for_version` is used to download a specific version (`spack fetch` and `spack install`)

`list_url` and `list_depth` are used to locate new versions (`spack versions` and `spack checksum`)

Both are needed in this case.

Comment (darmac) [reply var/spack/repos/builtin/packages/glusterfs/package.py]: Got it, thanks for your guidance.
Fix it in https://github.com/spack/spack/pull/15565/commits/5dc402712cbc5f30a64536c62b193154ddfd08b5

**Pull request merged**